### DISCUSSION TITLE
Model binding in mount() no work with nulls

### DISCUSSION THREAD

**Discussion by victore13** [livewire/livewire#7906] [7. Bugs (archived)]
### Livewire version

v3.4.4

### Laravel version

10.43.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. In the Livewire component class create a property of type `MyModel` or null (?MyModel) and `null` as default value.
2. In the `mount()` method add a parameter of type `MyModel` or null (?MyModel) and a default value as `null`.
3. When calling the component, don't add the parameter.

### Repository or Wirebox URL

https://wirebox.app/b/x5166

### Calling up the component in the view
The component is invoked, but parameters aren't passed to it.

```blade
<livewire:my-component />
``` 

### In the component class
The argument received in the mount() method is an object of type "MyModel" without attributes. I would expect to receive a null value in $myModel argument instead of an object of type "MyModel".

```php
class MyComponent extends Component
{
    public ?MyModel $myModel = null;

    public function mount(?MyModel $myModel): void
    {
        dd($myModel); //Object of type MyModel without attributes
        $this->myModel = $myModel;
    }
}
``` 

### Purpose
The purpose of this one is for the passing of the parameter is not mandatory.
I think that if the parameter is received in the mout() method, then would should force it to be of type that has been marked and if no parameter is received, then the argument would should be null and should not do the Model Binding.

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
its not bug. please search in google how dependency injection works on laravel.

here's my simple example:

if you're casting model class on parameter function like this:

```php
public function someFunction(?MyModel $myModel = null)
{
   dd($myModel);
}
```

it's same as:

```php
public function someFunction(?int $myModelId = null)
{
  $myModel = MyModel::findOrFail($myModelId) // so basically you're search model with id `null` and the result is 404 page
  dd($myModel);
}
```

here, i made a example in wirebox:
https://wirebox.app/b/x2z6r

**Reply by victore13** [reply]
Thanks you for reply!

I know how DI works in Laravel, but...
Why, if I expect a type 'MyModel' and I don't pass anything to it, instead of getting a 404 error, it returns a model with no attributes?

Like a this case: [Wirebox](https://wirebox.app/b/x9l5k)

_Attached screenshot_
![example](https://github.com/livewire/livewire/assets/73330191/8aa9acd3-8ac0-4922-832a-ba9d435ffcae)

**Reply by calebporzio** [reply]
Can you please submit a failing test PR?

**Reply by victore13** [reply]
After reviewing all the code flow I saw that the problem was in a Laravel's own static function that you use.

I have already opened the discussion in their project, let's see if they can give me some solution.

**Reply by joshhanley** [reply]
@victore13 Can you link to the discussion?

**Reply by victore13** [reply]
Of course @joshhanley ! 

Here it is: [Discussion](https://github.com/laravel/framework/discussions/50029)

**Comment by patie** [comment]
Its not only about models, even plain DTOs are confusing here...

when i do something like:
```
   public function mount(UserDTO $userDTO): void
    {
        dd($user);
        $this->user = $user;
        $this->form->fill();
    }
```

and in template i try call (without parameters) like:

```
        @livewire(\App\Livewire\Component\User\UserCreateForm::class)
```

`mount` method is completly ignored = method is not called, my excepted behaviour is component mounting fail, because of missing mount argument.. any way how to do it mandatory somehow? thank you
